lwm2m: Fixes for push mode FOTA (#15892)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * Updated to use a shorter resource string for the ``d2c/bulk`` resource.

* :ref:`lib_lwm2m_client_utils` library:

>   * Fixed an issue where failed delta update for the modem would not clear the state and blocks future delta updates. This only affected when LwM2M Firmware object is used in push mode.

Reviewer: ```suggestion
  * Fixed an issue where a failed delta update for the modem would not clear the state and blocks future delta updates.
    This only occurred when an LwM2M Firmware object was used in push mode.
```
